modules/contacts/client/components/Contact.component.js

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import ContactPresentational from './ContactPresentational';
+import RemoveContact from './RemoveContact';
+import * as contacts from '../api/contacts.api';
+
+const api = { contacts };
+
+export default class Contact extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showRemoveModal: false,
+      removeInProgress: false
+    };
+
+    this.handleRemoveContactModal = this.handleRemoveContactModal.bind(this);
+    this.handleRemoveContact = this.handleRemoveContact.bind(this);
+  }
+
+  handleRemoveContactModal(show) {
+    this.setState(() => ({ showRemoveModal: show }));
```

Comment:
Shouldn't this be:
```js
this.setState({showRemoveModal: show});
```

... unless I'm missing something?

It could even be shortened to:
```js
handleRemoveContactModal(showRemoveModal) {
  this.setState({showRemoveModal});
}
```

Reply:
#### Variable naming
:+1: 

#### Object vs functional updater

[Both are valid options.](https://reactjs.org/docs/react-component.html#setstate)

The issue with passing object to `this.setState` starts, when we pass object like `{ foo: this.state.foo + 1 }`. We should rather pass `(state) => ({ foo: state.foo + 1 })`.

When we don't reference `this.state` inside `this.setState`, both are fine. Functional updater is my personal preference based on a word of mouth.

Reply:
>Functional updater is my personal preference based on a word of mouth.

Generally speaking, if you don't pass it a function, you'll give React a chance to optimize setting the state (by letting it batch them) and potentially also improve performance if React doesn't need to re-render between each state-update. 

In situations where you need the previous state, you'll of course need the function to ensure the order of operation.

React is funky! 🙃

Reply:
As I am always using functional updaters I got curious about the React performance. :)  Do you have any source?
I was taught React try to batch them while passing functions too and official documentation didn't point any difference while talking about it https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous.

Reply:
@agatatalita I believe this bit from the docs had me believe that render happens in between synchronous state updates:

>In React, both this.props and this.state represent the rendered values, i.e. what’s currently on the screen.

via https://reactjs.org/docs/faq-state.html

but I'll do more reading and/or testing. :-)

Reply:
Changed the variable naming in 15f8cba5a99783deab9b254792a76a43bb56f99b.

The `setState` behaviour still seems to be unresolved.

Reply:
The [`setState` in React's source](https://github.com/facebook/react/blob/master/packages/react/src/ReactBaseClasses.js#L58-L67). It is passing both the updater object and function to `this.updater.enqueueSetState()`. What is the `this.updater` there?

From the [source code](https://github.com/facebook/react/blob/master/packages/react/src/ReactBaseClasses.js#L26-L27):
> We initialize the default updater but the real one gets injected by the renderer.

Here I lost the thread. We'd have to find the real updater to see how it treats objects and functions passed to `setState`.